Jenkinsfile.deploy

```diff
@@ -196,9 +209,23 @@ pipeline {
 
         stage('Deploy artifacts') {
             steps {
-                mavenDeploy('optaplanner', '-Dfull')
-                mavenDeploy('optaweb-vehicle-routing', "-s ${env.GLOBAL_MAVEN_SETTINGS}")
-                mavenDeploy('optaweb-employee-rostering', "-s ${env.GLOBAL_MAVEN_SETTINGS}")
+                script {
+                    if(!isRelease() || params.MAVEN_DEPLOY_REPOSITORY){
+                        // Normal deploy
+                        runMavenDeploy(defaultOptaplannerMvnCmd)
+                        runMavenDeploy(defaultOptawebVehicleRoutingMvnCmd)
+                        runMavenDeploy(defaultOptawebEmployeeRosteringMvnCmd)
+                    } else {
+                        // Deploy locally and then to staging
+                        runMavenDeployLocally(defaultOptaplannerMvnCmd, optaplannerRepository)
+                        runMavenDeployLocally(defaultOptawebVehicleRoutingMvnCmd, vehicleRoutingRepository)
+                        runMavenDeployLocally(defaultOptawebEmployeeRosteringMvnCmd, employeeRosteringRepository)
+
+                        runMavenStage(defaultOptaplannerMvnCmd, optaplannerRepository)
+                        runMavenStage(defaultOptawebVehicleRoutingMvnCmd, vehicleRoutingRepository)
+                        runMavenStage(defaultOptawebEmployeeRosteringMvnCmd, employeeRosteringRepository)
```

Comment:
The `if-else` can be replaced by a method, e.g. `deployMavenArtifacts()` that would take care of a single repository.

Reply:
something like:
```
deployMavenArtifacts(defaultOptaplannerMvnCmd, optaplannerRepository)
deployMavenArtifacts(defaultOptawebVehicleRoutingMvnCmd, vehicleRoutingRepository)
deployMavenArtifacts(defaultOptawebEmployeeRosteringMvnCmd, employeeRosteringRepository)
```
and then
```
void deployMavenArtifacts(mvnCmd, directory){
    if(!isRelease() || params.MAVEN_DEPLOY_REPOSITORY){
         runMavenDeploy(mvnCmd)
    } else {
        runMavenDeployLocally(mvnCmd, directory)
        runMavenStage(mvnCmd, directory)
    }
}
```
?